Plugin crashed when NAT not found

### Terraform Version
<!---
Run `terraform version` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
terraform-provider-outscale_v0.8.2
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For Terraform configs larger than a few resources, or that involve multiple files, please make a GitHub repository that we can clone, rather than copy-pasting multiple files in here.
-->

```terraform
available at OSC internal gitlab repo, ask the issuer to have access.
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly.
-->

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->
```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Request cancelled
│
│   with outscale_nat_service.nat01,
│   on nat.tf line 1, in resource "outscale_nat_service" "nat01":
│    1: resource "outscale_nat_service" "nat01" {
│
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵
╷
│ Error: Plugin did not respond
│
│   with outscale_vm.vm01,
│   on vm.tf line 1, in resource "outscale_vm" "vm01":
│    1: resource "outscale_vm" "vm01" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-outscale_v0.8.2 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 402 [running]:
github.com/terraform-providers/terraform-provider-outscale/outscale.resourceOAPINatServiceRead.func1(0xc000847a80)
        github.com/terraform-providers/terraform-provider-outscale/outscale/resource_outscale_nat_service.go:182 +0x507
github.com/terraform-providers/terraform-provider-outscale/outscale.resourceDataAttrSetter(0xc0009607e0, 0xc000b838f0)
        github.com/terraform-providers/terraform-provider-outscale/outscale/resource_outscale_vm.go:1334 +0x62
github.com/terraform-providers/terraform-provider-outscale/outscale.resourceOAPINatServiceRead(0xc0009607e0, {0xff0c00?, 0xc000013aa8?})
        github.com/terraform-providers/terraform-provider-outscale/outscale/resource_outscale_nat_service.go:161 +0x567
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00012d4a0, 0xc000181810, {0xff0c00, 0xc000013aa8})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:470 +0x1aa
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000112368, {0xc000e08360?, 0x4ba926?}, 0xc000e08360)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:535 +0x365
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler({0x11a5e60?, 0xc000112368}, {0x1512070, 0xc000b1bd70}, 0xc000e08300, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3269 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003e9a40, {0x1516680, 0xc00043e600}, 0xc000120300, 0xc0008c8de0, 0x1d226d0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0xcab
google.golang.org/grpc.(*Server).handleStream(0xc0003e9a40, {0x1516680, 0xc00043e600}, 0xc000120300, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x28a

Error: The terraform-provider-outscale_v0.8.2 plugin crashed!
```

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->
terraform init  __uses a gitlab backend holding the state files (that should not change behavior)__
terraform plan
terraform apply
Terraform creates all the resources, NAT included
DELETE NAT BY HAND
Re-launch terraform plan
error appears

### Expected Behavior
<!--
What should have happened?
-->
Terraform should find out there is a missing NAT and add the resource to the plan
The resource should be re-created

### Actual Behavior
<!--
What actually happened?
-->
Provider plugin crashed !

### Additional Context
<!--
Are there anything atypical about your situation that we should know?
-->

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? 
-->

--- a/outscale/resource_outscale_nat_service.go
+++ b/outscale/resource_outscale_nat_service.go
@@ -139,19 +139,18 @@ func resourceOAPINatServiceRead(d *schema.ResourceData, meta interface{}) error 
 		Filters: &oscgo.FiltersNatService{NatServiceIds: &[]string{d.Id()}},
 	}
 
-	stateConf := &resource.StateChangeConf{
-		Pending: []string{"pending"},
-		Target:  []string{"available", "deleted"},
-		Refresh: NGOAPIStateRefreshFunc(conn, filterReq, "failed"),
-		Timeout: 10 * time.Minute,
-	}
-
-	value, err := stateConf.WaitForState()
+	var resp oscgo.ReadNatServicesResponse
+	err := resource.Retry(120*time.Second, func() *resource.RetryError {
+		rp, httpResp, err := conn.NatServiceApi.ReadNatServices(context.Background()).ReadNatServicesRequest(filterReq).Execute()
+		if err != nil {
+			return utils.CheckThrottling(httpResp, err)
+		}
+		resp = rp
+		return nil
+	})
 	if err != nil {
 		return fmt.Errorf("error waiting for NAT Service (%s) to become available: %s", d.Id(), err)
 	}
-
-	resp := value.(oscgo.ReadNatServicesResponse)
 	if utils.IsResponseEmpty(len(resp.GetNatServices()), "NatService", d.Id()) {
 		d.SetId("")
 		return nil

